icecube_tools/point_source_likelihood/point_source_likelihood.py

```
@@ -618,6 +720,165 @@ def get_test_statistic(self):
+
+        Uses the iMinuint wrapper.
+        """
+
+        init_index = 2.19  # self._energy_likelihood._min_index + (self._max_index - self._energy_likelihood._min_index)/2
+        error_index = 0.1
+        some_llh = self.likelihoods[list(self.likelihoods.keys())[0]]
+        limit_index = (some_llh._energy_likelihood._min_index,
+            some_llh._energy_likelihood._max_index)
+        # Get init_ns and limit_ns for each period
+        # could be nicer with some generator method
+        init = []
+        limits = []
+        for llh in self.likelihoods.values():
+            init.append(llh._ns_min + (llh._ns_max - llh._ns_min) / 2)
+            limits.append((llh._ns_min, llh._ns_max))
+
+        # Get errors to start with
+        errors = [1 for _ in init]  
+        name = tuple(f"n{i}" for i in range(len(init)))  
+
+        if self.which == 'spatial':
+            #Only spatial-only likelihood needs special function, because no spectral index is used
+            func_to_minimize = self._func_to_minimize_sp
+        else:
+            func_to_minimize = self._func_to_minimize
+            #add errors, limits, start value and name of index
+            errors += [error_index]
+            limits += [limit_index]
+            init += [init_index]
+            name += ("index",)
+
+        m = Minuit(func_to_minimize, *init, name=name)
+        m.errordef = 0.5
+        m.errors = errors
+        m.limits = limits
+        print(m.limits)
+        print(m.errors)
+        print(m.values)
+        """
+        if self.which == 'spatial':
+            m.fixed["index"] = True
+            m.values["index"] = 2.
+        """
+        
+        m.migrad()
+        if self.which != 'spatial':
+            if not m.fmin.is_valid or not m.fmin.has_covariance:
+
+                # Fix the index as can be uninformative
+                m.fixed["index"] = True
+                m.migrad()
+        else:
+            pass
+            #what else to do?
```

Maybe raise an error or warning here?

---

good idea, will do so